### PR TITLE
[BEAM-1279] Ensure Composite Nodes produce no output

### PR DESCRIPTION
**Pull request by tgroh**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by tgroh** [comment]
R: @lukecwik

File: sdks/java/core/src/main/java/org/apache/beam/sdk/runners/TransformHierarchy.java [hunk 1]
```diff
@@ -290,22 +290,40 @@ private void setOutput(POutput output) {
...
-                    + "outputs of a composite transform must be produced by a component transform "
-                    + "or be part of the input."
-                    + "%n    Other Outputs: %s"
-                    + "%n    Other Producers: %s",
-                getFullName(), POutput.class.getSimpleName(), output.expand(), otherProducerNames));
+        if (!parts.isEmpty()) {
+          throw new IllegalArgumentException(
+              String.format(
+                  "Output of transform [%s] contains a %s produced by it, but it is a composite "
+                      + "Transform. Output of composite transforms must be produced by component "
```
Comment (lukecwik) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/runners/TransformHierarchy.java]: Can we collapse both error messages to the single following statement?
Output of composite transform [%s] contains a primitive %s produced by it. Only primitive transforms are allowed to produce primitive outputs.
%n   Outputs:
%n   Nodes:

Comment (tgroh) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/runners/TransformHierarchy.java]: Done.

**Comment by lukecwik** [comment]
LGTM

**Pull request closed without merging**